File: src/main/java/cz/vhromada/catalog/web/mapper/MusicMapper.java
package cz.vhromada.catalog.web.mapper;

import cz.vhromada.catalog.entity.Music;
import cz.vhromada.catalog.web.fo.MusicFO;

import org.mapstruct.Mapper;

/**
 * An interface represents mapper for music.
 *
 * @author Vladimir Hromada
 */
@Mapper
public interface MusicMapper {

    /**
     * Returns FO for music.
     *
     * @param source music
     * @return FO for music
     */
    MusicFO map(Music source);

    /**
     * Returns music.
     *
     * @param source FO for music
     * @return music
     */
    Music mapBack(MusicFO source);

}

File: src/main/kotlin/cz/vhromada/catalog/web/mapper/MusicMapper.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.entity.Music
import cz.vhromada.catalog.web.fo.MusicFO

/**
 * An interface represents mapper for music.
 *
 * @author Vladimir Hromada
 */
interface MusicMapper {

    /**
     * Returns FO for music.
     *
     * @param source music
     * @return FO for music
     */
    fun map(source: Music): MusicFO

    /**
     * Returns music.
     *
     * @param source FO for music
     * @return music
     */
    fun mapBack(source: MusicFO): Music

}
